### PR TITLE
Add config param to pass additional parameters to confluent-kafka-python

### LINKED ISSUE #1405: Problems configuring faststream for SASL_SSL security
### Discussed in https://github.com/airtai/faststream/discussions/1404

<div type='discussions-op-text'>

<sup>Originally posted by **jorrgme** April 29, 2024</sup>
Hi all,

I really need your help, as I've been trying everything to be able to consume from a topic using SASL_SSL authentication.

This is the code I'm using:

```python
import ssl
from faststream import FastStream
from faststream.confluent import KafkaBroker
from faststream.security import SASLPlaintext

user      = "my_usr"
passw     = "my_pass"
bootstrap = "my.bootstrap.server.com:443"
cert      = "~/my_certs/CRT_cacerts.pem"
group     = "my_group"
topic     = "my.topic.cpd"

ssl_context = ssl.create_default_context(
    cafile=cert,
)
security = SASLPlaintext(
    ssl_context=ssl_context,
    username=user,
    password=passw,
    use_ssl=True
)

broker = KafkaBroker(
    bootstrap_servers=bootstrap,
    security=security,
)

app = FastStream(broker)

@broker.subscriber(topic, group_id=group)
async def handle_msg(data) -> str:
    print(data)
```

And this is the log with ssl verifying errors I'm getting:

```
2024-04-29 10:41:59,089 INFO     - FastStream app starting...
%3|1714380119.145|FAIL|faststream-0.5.2#producer-1| [thrd:sasl_ssl://my.bootstrap.server.com:443/bootstrap]: sasl_ssl://my.bootstrap.server.com:443/bootstrap: SSL handshake failed: error:0A000086:SSL routines::certificate verify failed: broker certificate could not be verified, verify that ssl.ca.location is correctly configured or root CA certificates are installed (brew install openssl) (after 9ms in state SSL_HANDSHAKE)
```

Does anyone have any idea of how I can solve my issues?

I've tried the same thing using `Faust` and it's working fine using the same certificate, user and password.

Thanks in advance for any help you can provide.</div>

### PR DESCRIPTION
**Pull request by kumaranvpl**
# Description

- [x] Add `config` param which needs to be passed to `confluent-kafka-python`
- [x] Remove `ssl_context`
- [x] Raise `SetupError` if `ssl_context` is passed
- [x] Use `TypedDict` for `config`
- [x] Remove unused params from Async wrapper which are added to maintain feature parity with AIOKafka implementation. This is not needed anymore.

Fixes #1405 

## Type of change

Please delete options that are not relevant.

- [x] Documentation (typos, code examples, or any documentation updates)
- [x] Bug fix (a non-breaking change that resolves an issue)
- [x] New feature (a non-breaking change that adds functionality)
- [x] Breaking change (a fix or feature that would disrupt existing functionality)
- [x] This change requires a documentation update

## Checklist

- [x] My code adheres to the style guidelines of this project (`scripts/lint.sh` shows no errors)
- [ ] I have conducted a self-review of my own code
- [x] I have made the necessary changes to the documentation
- [x] My changes do not generate any new warnings
- [ ] I have added tests to validate the effectiveness of my fix or the functionality of my new feature
- [x] Both new and existing unit tests pass successfully on my local environment by running `scripts/test-cov.sh`
- [x] I have ensured that static analysis tests are passing by running `scripts/static-analysis.sh`
- [x] I have included code examples to illustrate the modifications

### CODE REVIEW

**Comment by kumaranvpl** [comment]
@Lancetnik Points to discuss

1. I am removing `ssl_context` because it is causing confusion and `confluent-kafka-python` can't use `ssl_context` directly. It uses separate individual security params. Shall I raise error if `ssl_context` is passed ? - https://github.com/airtai/faststream/blob/pass-config-to-confluent/faststream/confluent/security.py#L16

2. Confluent kafka python accepts huge list of config mentioned here - https://github.com/confluentinc/librdkafka/blob/master/CONFIGURATION.md. To make users' life easier we have added some of those config as producer/consumer [params](https://github.com/airtai/faststream/blob/pass-config-to-confluent/faststream/confluent/client.py#L87-L110), rest can be sent as part of [config dict](https://github.com/airtai/faststream/blob/pass-config-to-confluent/faststream/confluent/client.py#L111). When we are merging params and config dict which should take precedence? params or config items? - https://github.com/airtai/faststream/blob/pass-config-to-confluent/faststream/confluent/client.py#L151

**Comment by Lancetnik** [comment]
@kumaranvpl your decision looks good for me, but I have some notices:

1) We should raise RuntimeWarning if user setted up `ssl_context` option in `parse_security` method instead of suppressing it

2) We should created a TypedDict for `config` option to make it auto-compliteable and much explicit. Also, it should solves problems with another init parameters conflict (but the current merging priority looks good too)

File: faststream/confluent/security.py [hunk 1]
```diff
@@ -14,6 +15,9 @@
 
 
 def parse_security(security: Optional[BaseSecurity]) -> "AnyDict":
+    if security and isinstance(security.ssl_context, ssl.SSLContext):
```
Comment (Lancetnik) [inline faststream/confluent/security.py:18]: I think, we should use `SetupError` instead of basic `ValueError` in wrong broker parameters case

Comment (kumaranvpl) [reply faststream/confluent/security.py:18]: Noted. Thanks.

**Review by Lancetnik — APPROVED** [review]

**Review by Lancetnik — DISMISSED** [review]
Lgtm, I think, we can release it

**Review by Lancetnik — APPROVED** [review]

**Pull request merged**